src/Subscriptions.ts

```diff
@@ -100,11 +107,13 @@ export function updateCurrentSubscriptionPeriod(module: Address, timestamp: BigI
   subscriptionsModule.currentPeriod = periodId
   subscriptionsModule.save()
 
+  let currentPeriod = subscriptions.getPeriod(periodId)
   let period = loadOrCreateSubscriptionPeriod(periodId, timestamp)
   period.instance = module.toHex()
   period.feeToken = subscriptions.currentFeeToken()
   period.feeAmount = subscriptions.currentFeeAmount()
```

Comment:
These shouldn't be fetched from the current period instead?

Reply:
Sure! I missed that!